app_flutter/lib/status_grid.dart

```diff
@@ -99,64 +99,152 @@ class StatusGrid extends StatelessWidget {
   /// Reference to the build state to perform actions on [TaskMatrix], like rerunning tasks.
   final FlutterBuildState buildState;
 
+  @override
+  _StatusGridState createState() => _StatusGridState();
+}
+
+class _StatusGridState extends State<StatusGrid> {
+  List<ScrollController> controllers;
+  SyncScrollController _syncScroller;
+
   @override
   Widget build(BuildContext context) {
-    /// The grid needs to know its dimensions. Column is based off how many tasks are
-    /// in a row (+ 1 to account for [CommitBox]).
-    final int columnCount = taskMatrix.columns + 1;
+    controllers = List<ScrollController>.generate(
```

Comment:
Can you generate these controllers in initState and only update them when the matrix changes?

Reply:
Done.

Since the matrix will always have the same amount of rows for now, we don't need to worry about adding more scroll controllers if more rows are updated.